src/server.rs

```diff
@@ -513,6 +583,45 @@ mod tests {
         Ok(())
     }
 
+    #[actix_web::test]
+    async fn test_list() -> Result<(), Error> {
+        let mut config = Config::default();
+        config.server.expose_list = Some(true);
+
+        let test_upload_dir = "test_upload";
+        fs::create_dir(test_upload_dir)?;
+        config.server.upload_path = PathBuf::from(test_upload_dir);
+
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::new(RwLock::new(config)))
+                .app_data(Data::new(Client::default()))
+                .configure(configure_routes),
+        )
+        .await;
+
+        let file_name = "test_file.txt";
+        let timestamp = util::get_system_time()?.as_secs().to_string();
+        test::call_service(
+            &app,
+            get_multipart_request(&timestamp, "file", file_name).to_request(),
+        )
+        .await;
+
+        let request = TestRequest::default()
+            .insert_header(("content-type", "text/plain"))
+            .uri("/list")
+            .to_request();
+        let result: Vec<ListItem> = test::call_and_read_body_json(&app, request).await;
+
+        assert_eq!(result.len(), 1);
+        assert!(result.first().is_some());
```

Comment:
assert_eq!(result.first().expect("json object").file_name, file_name);
```suggestion
        assert!(result.first().is_some());
```

A panic is perfectly fine here. `assert` also panics, if the condition is not met.

The issue was that clippy complained about unwrap, which in my opinion should be ignored by clippy for tests.

To get rid of the message there are 2 options:

- allow unwrap for clippy for that test
- use `expect` instead

Reply:
Thank you! This is where my lack of Rust knowledge bites me... I think the `expect` syntax is simple enough.

Reply:
Same here. I am always happy when orhun shows me the Rust way of doing things...